references/classification/WildFire/torch/train.py

```diff
@@ -0,0 +1,310 @@
+#!usr/bin/python
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import pandas as pd
+import math
+from pathlib import Path
+import os
+import random
+from torch.utils.data import DataLoader
+import torch
+from torch import nn
+from torch import optim
+import torchvision
+from torchvision import transforms
+from fastprogress import master_bar, progress_bar
+import torch.nn.functional as F
+from torch.autograd import Variable
+import torch.utils.data as data
+from torch.utils.data.sampler import Sampler
+from torch.utils.data.dataset import Dataset
+from torch.optim.lr_scheduler import OneCycleLR
+from pyronear import models
+from pyronear.datasets.wildfire import (WildFireDataset, WildFireSplitter, FireLabeler)
+from pyronear.datasets.wildfire.split_strategy import *
+
+
+def set_seed(seed):
+    """Set the seed for pseudo-random number generations
+    Args:
+        seed (int): seed to set for reproducibility
+    """
+
+    random.seed(seed)
+    os.environ['PYTHONHASHSEED'] = str(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)  # if you are using multi-GPU.
+    torch.backends.cudnn.benchmark = False
+    torch.backends.cudnn.deterministic = True
+
+
+def train_batch(model, x, target, optimizer, criterion):
```

Comment:
Does target needs to be squeeze() here ?

Reply:
Not sure why it would? The model will need the batch dimension, so we only `squeeze` it if it was already `unsqueeze`. If I'm not mistaking, it's not needed then

Reply:
It depend of your dataloader, but in this script there is no need to squeeze the target